src/client/components/pages/Courses/CoursesPage.tsx

```diff
@@ -4,36 +4,46 @@ import React, {
   useState,
   useEffect,
   useContext,
+  useMemo,
 } from 'react';
-import { LoadSpinner } from 'mark-one';
+import { Dropdown, LoadSpinner, POSITION } from 'mark-one';
 import { MessageContext } from 'client/context';
 import CourseInstanceResponseDTO from 'common/dto/courses/CourseInstanceResponse';
 import { CourseAPI } from 'client/api';
 import { MESSAGE_TYPE, MESSAGE_ACTION, AppMessage } from 'client/classes';
 import { OFFERED, COURSE_TABLE_COLUMN } from 'common/constants';
+import { ViewResponse } from 'common/dto/view/ViewResponse.dto';
 import CourseInstanceTable from './CourseInstanceTable';
 import { tableFields } from './tableFields';
 
-/*
- * TODO
- * Until the functionality for defining a retrieving custom view is implemented
- * we can just hard code/comment the columns to display
+/**
+ * These columns are ALWAYS shown regardless of user choice
  */
-const currentView = [
+const mandatoryColumns = [
   COURSE_TABLE_COLUMN.AREA,
   COURSE_TABLE_COLUMN.CATALOG_NUMBER,
-  COURSE_TABLE_COLUMN.TITLE,
-  COURSE_TABLE_COLUMN.SAME_AS,
-  COURSE_TABLE_COLUMN.IS_SEAS,
-  // COURSE_TABLE_COLUMN.IS_UNDERGRADUATE,
-  COURSE_TABLE_COLUMN.OFFERED,
-  COURSE_TABLE_COLUMN.INSTRUCTORS,
-  COURSE_TABLE_COLUMN.MEETINGS,
-  // COURSE_TABLE_COLUMN.ENROLLMENT,
   COURSE_TABLE_COLUMN.NOTES,
   COURSE_TABLE_COLUMN.DETAILS,
 ];
 
+/**
+ * Default View
+ *
+ * This is the hard-coded default view that is showin in the dropdown when a
+ * user loads the page. Other views can be created, modified and deleted - but
+ * this default view can never be deleted or updated.
+ */
+const defaultView: ViewResponse = {
+  id: 'default',
+  name: 'Default',
+  columns: [
+    ...mandatoryColumns,
+    COURSE_TABLE_COLUMN.IS_SEAS,
+    COURSE_TABLE_COLUMN.OFFERED,
+    COURSE_TABLE_COLUMN.INSTRUCTORS,
```

Comment:
Good idea. Fixed that in a84ae36f2cd15cfb9e2cf69b09c47987041a19b9.


Thinking about this, I'm not sure if we _do_ need to update those tests do we...since the default view will always be the default.

Reply:
Well, it depends on whether we want to make meetings a permanent part of the the Default view, since it isn't in the existing course planner. I think that's something we'd want to talk though with the users though.

Reply:
Gonna add a reference to this discussion in the issue.